github/repos_contents.go

```diff
@@ -91,6 +93,38 @@ func (s *RepositoriesService) GetReadme(owner, repo string, opt *RepositoryConte
 	return readme, resp, err
 }
 
+// DownloadContents can be used to download files.  It circumvents the normal
```

Comment:
I don't think the details of exactly how this is implemented is important for the public documentation here.  Maybe just something like:

> DownloadContents returns an io.ReadCloser that reads the contents of the specified file.  This function will work with files of any size, as opposed to GetContents which is limited to 1 Mb files.  It is the caller's responsibility to close the ReadCloser.